pootle/core/models/snapshot.py

```diff
@@ -90,8 +92,10 @@ def save(self, ctx_data):
         except OSError:
             pass
 
-        with open(self.filepath, 'w') as outfile:
-            outfile.write(ctx_data)
+        # use newline='' not to convert \n to OS-specific newlines;
+        # this is to be able to generate snapshots on any platform
+        with io.open(self.filepath, 'w', newline='') as outfile:
+            outfile.write(unicode(ctx_data))
```

Comment:
Can you provide a reference perhaps? I cannot comment much without really knowing what you are referring to :)

The only thing I am aware of is that in Python 3 there would be no need to use the `io` module, since the built-in `open()` function accepts the `newline` parameter.

For this particular line, the point is that there is no need to perform extra conversions from the `str` type (which are actual bytes, and this is represented via the `bytes` type in Python 3) to the `unicode` text type (represented by the `str` type in Python 3). The data is ready to be directly written to disk.

Generally speaking, I'm not concerned about Python 3 so long as it doesn't offer compelling reasons to upgrade (and this is not one of them). Making code future-proof if it comes for free is 👍 though.